tuf/src/metadata.rs

```diff
@@ -435,6 +435,7 @@ where
         self.signatures
             .retain(|s| s.key_id() != private_key.public().key_id());
         self.signatures.push(sig);
+        self.signatures.sort();
```

Comment:
Were we just getting lucky with this before?

Reply:
Not necessarily. The order of signatures was entirely dependent on the order you called `SignedMetadata::add_signature`, so things were consistent if we stored the keys in an ordered list, which we've done in the tests. This change just makes sure clients don't run into non-deterministic output if they happen to sign things out of order.